syntax = "proto3";
package sqlpkg_versions;





/*
ListSQLPackageVersion请求
*/
message ListSQLPackageVersionRequest {
    /*
    SQL包ID
    */
    string pkgId = 1 ;
    /*
    页码
    */
    int32 page = 2 ;
    /*
    每页数据量
    */
    int32 pageSize = 3 ;
}

/*
ListSQLPackageVersion返回
*/
message ListSQLPackageVersionResponse {
message List {
message SqlPackage {
    /*
    SQL包ID
    */
    string id = 1 ;
    /*
    名称
    */
    string name = 2 ;
    /*
    描述
    */
    string memo = 3 ;
    /*
    文件仓库包ID
    */
    string repoPackageId = 4 ;
    /*
    创建者
    */
    string creator = 5 ;
    /*
    mtime
    */
    int64 mtime = 6 ;
}
        
    
    /*
    所属的SQL包
    */
    SqlPackage sqlPackage = 1 ;
    /*
    版本ID
    */
    string id = 2 ;
    /*
    名称
    */
    string name = 3 ;
    /*
    仓库版本ID
    */
    string repoVersionId = 4 ;
    /*
    程序包版本ID
    */
    string appVersionId = 5 ;
    /*
    版本名称
    */
    string versionName = 6 ;
    /*
    备注
    */
    string memo = 7 ;
    /*
    创建者
    */
    string creator = 8 ;
    /*
    创建时间
    */
    int64 ctime = 9 ;
    /*
    修改时间
    */
    int64 mtime = 10 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListSQLPackageVersionApi返回
*/
message ListSQLPackageVersionResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    sqlpkg_versions.ListSQLPackageVersionResponse data = 4 ;
}

